wrappers/python/tests/layer/test_layer_norm.py

```diff
@@ -19,149 +21,194 @@
-def test_layer_norm(dtype: np.dtype):
-    # Describe single-tile tensor, located at node 0
-    A_shape = [20, 30]
-    ndim = len(A_shape)
-    eps = 1e-5
-    A_traits = nntile.tensor.TensorTraits(A_shape, A_shape)
-    mpi_distr = [0]
-    next_tag = 0
-    # Tensor objects
-    A_value = Tensor[dtype](A_traits, mpi_distr, next_tag)
-    next_tag = A_value.next_tag
-    A_grad = Tensor[dtype](A_traits, mpi_distr, next_tag)
-    next_tag = A_grad.next_tag
-    A = nntile.tensor.TensorMoments(A_value, A_grad, True)
-    # Set initial values of tensors
+single_tile = LayerNormTestParams(
+    n_size=10,
+    n_size_dyn=20,
+    n_size_tile=10,
+    m_size=30,
+    m_size_tile=30,
+)
+
+multiple_tiles = LayerNormTestParams(
+    n_size=10,
+    n_size_dyn=20,
+    n_size_tile=5,
+    m_size=30,
+    m_size_tile=15,
+)
+
+
+def generate_inputs(dtype: str, params: LayerNormTestParams):
     rng = np.random.default_rng(42)
-    rand_A = rng.standard_normal(A_shape)
-    np_A = np.array(rand_A, dtype=dtype, order='F')
-    A.value.from_array(np_A)
-    torch_A = torch.tensor(np_A, requires_grad=True)
-    rand_B_grad = rng.standard_normal(A_shape)
-    np_B_grad = np.array(rand_B_grad, dtype=dtype, order='F')
-    rand_gamma = rng.standard_normal(A_shape[-1])
-    np_gamma = np.array(rand_gamma, dtype=dtype, order='F')
-    rand_beta = rng.standard_normal(A_shape[-1])
-    np_beta = np.array(rand_beta, dtype=dtype, order='F')
-    # Init NNTile LayerNorm
-    nntile_layer, next_tag = nntile.layer.LayerNorm \
-        .generate_simple(A, ndim - 1, eps, next_tag)
-    nntile_layer.gamma.value.from_array(np_gamma)
-    nntile_layer.beta.value.from_array(np_beta)
-    # Init PyTorch LayerNorm
-    torch_layer = LayerNorm(A_shape[-1], eps=eps, dtype=torch_A.dtype)
+    eps = 1e-05
+
+    torch_layer = LayerNorm(params.m_size, eps=eps)
+    rand_gamma = rng.standard_normal(params.m_size)
+    np_gamma = np.array(rand_gamma, dtype=np.float32, order='F')
+    rand_beta = rng.standard_normal(params.m_size)
+    np_beta = np.array(rand_beta, dtype=np.float32, order='F')
     torch_layer.weight.data = torch.tensor(np_gamma)
     torch_layer.bias.data = torch.tensor(np_beta)
-    # NNTile forward of LayerNorm
-    nntile_layer.forward_async()
-    np_B_nntile = np.zeros_like(np_A)
-    nntile_layer.y.value.to_array(np_B_nntile)
-    # PyTorch forward
-    torch_B = torch_layer(torch_A)
-    np_B_torch = torch_B.data.numpy()
-    # Check forward
-    abs_error = np.linalg.norm(np_B_torch - np_B_nntile)
-    assert abs_error / np.linalg.norm(np_B_torch) <= 1e-5
-    # NNTile backward of LayerNorm
-    nntile_layer.y.grad.from_array(np_B_grad)
+
+    x_shape = [params.n_size, params.m_size]
+    x_basetile = [params.n_size_tile, params.m_size_tile]
+    x_traits = TensorTraits(x_shape, x_basetile)
+    x_distr = [0] * x_traits.grid.nelems
+    x_type = dtype2nntile[dtype]
+    x_value = x_type(x_traits, x_distr, 0)
+    x_grad = x_type(x_traits, x_distr, 0)
```

Comment:
make `x_grad=zeros_like(x_value)`. `zeros_like` is in `nntile.utils.constructors`